gatsby-plugin-netlify-cms does not work with netlifycms v2 + gatsby v1

Error: ./~/gatsby-plugin-netlify-cms/cms.js
  Module not found: Error: Cannot resolve module 'netlify-cms/dist/cms.css' in C:\Users\ap\.root\ws\gm\node_modules\gatsby-  plugin-netlify-cms
  resolve module netlify-cms/dist/cms.css in C:\Users\ap\.root\ws\gm\node_modules\gatsby-plugin-netlify-cms
    looking for modules in C:\Users\ap\.root\node_modules
      C:\Users\ap\.root\node_modules\netlify-cms doesn't exist (module as directory)
    looking for modules in C:\Users\ap\.root\ws\gm\node_modules\gatsby-plugin-netlify-cms\node_modules
      C:\Users\ap\.root\ws\gm\node_modules\gatsby-plugin-netlify-cms\node_modules\netlify-cms doesn't exist (module as dire  ctory)

 Error: ./~/netlify-cms/dist/netlify-cms.js
  Module parse failed: C:\Users\ap\.root\ws\gm\node_modules\netlify-cms\dist\netlify-cms.js Unexpected token (14:144353)
  You may need an appropriate loader to handle this file type.
  SyntaxError: Unexpected token (14:144353)
      at Parser.pp$4.raise (C:\Users\ap\.root\ws\gm\node_modules\acorn\dist\acorn.js:2221:15)

--- a/packages/gatsby-plugin-netlify-cms/README.md
+++ b/packages/gatsby-plugin-netlify-cms/README.md
@@ -1,5 +1,11 @@
 # gatsby-plugin-netlify-cms
 
+**Gatsby v1 and Netlify CMS 1.x require [`gatsby-plugin-netlify-cms@^2.0.0`](https://github.com/gatsbyjs/gatsby/blob/gatsby-plugin-netlify-cms@2.0.1/packages/gatsby-plugin-netlify-cms/README.md).**
+
+**Gatsby v2 and Netlify CMS 2.x require `gatsby-plugin-netlify-cms@^3.0.0-beta.0`, which is documented below.**
+
+## Overview
+
 Automatically generates an `admin/index.html` with a default Netlify CMS implementation.
 
 Netlify CMS is a React single page app for editing git based content via API.
@@ -10,7 +16,7 @@ site](https://netlifycms.org).
 ## Install
 
 ```shell
-npm install --save netlify-cms gatsby-plugin-netlify-cms
+npm install --save netlify-cms gatsby-plugin-netlify-cms@next
 ```
 
 ## How to use
@@ -40,7 +46,9 @@ widgets](https://www.netlifycms.org/docs/custom-widgets/#registerwidget) or
 styling the [preview
 pane](https://www.netlifycms.org/docs/customization/#registerpreviewstyle),
 you'll need to do so in a JavaScript module and provide Gatsby with the path to
-your module via the `modulePath` option:
+your module via the `modulePath` option. Any styles imported by this module (or
+by the modules that it imports, all the way down the chain) are automatically
+applied to the editor preview pane by the plugin.
 
 ```javascript
 plugins: [
@@ -60,7 +68,22 @@ plugins: [
 The js module might look like this:
 
 ```javascript
+/**
+ * The default export of `netlify-cms` is an object with all of the Netlify CMS
+ * extension registration methods, such as `registerWidget` and
+ * `registerPreviewTemplate`.
+ */
 import CMS from `netlify-cms`
+
+/**
+ * Any imported styles will automatically be applied to the editor preview
+ * pane, there is no need to use `registerPreviewStyle` for imported styles.
+ * All of the example imports below would result in styles being applied to the
+ * preview pane.
+ */
+import `module-that-imports-styles.js`
+import `styles.scss`
+import `../other-styles.css`
 
 /**
  * Let's say you've created widget and preview components for a custom image
@@ -95,21 +118,6 @@ plugins: [
   },
 ]
 ```
-
-### `stylesPath`
-
-(_optional_, default: `undefined`)
-
-Gatsby template components can be used as [preview
-templates](https://www.netlifycms.org/docs/customization/) in Netlify CMS. To
-apply your site styles to the preview pane as well, you would normally register
-a [custom
-stylesheet](https://www.netlifycms.org/docs/customization/#registerpreviewstyle),
-but your Gatsby style source may be Sass or CSS modules, and can't be passed to
-Netlify CMS as is. The `stylesPath` accepts a path or an array of paths to your
-raw styles. The styles are built using the same Webpack and Babel configuration
-that your Gatsby site uses, and the CSS output is automatically registered and
-used in the preview pane.
 
 ### `publicPath`
 

--- a/packages/gatsby-plugin-netlify-cms/package.json
+++ b/packages/gatsby-plugin-netlify-cms/package.json
@@ -1,13 +1,14 @@
 {
   "name": "gatsby-plugin-netlify-cms",
   "description": "A Gatsby plugin which generates the Netlify CMS single page app",
-  "version": "2.0.0-beta.7",
+  "version": "3.0.0-beta.0",
   "author": "Shawn Erquhart <shawn@erquh.art>",
   "bugs": {
     "url": "https://github.com/gatsbyjs/gatsby/issues"
   },
   "dependencies": {
     "friendly-errors-webpack-plugin": "^1.7.0",
+    "html-webpack-exclude-assets-plugin": "^0.0.7",
     "html-webpack-plugin": "^3.2.0",
     "lodash": "^4.17.10",
     "mini-css-extract-plugin": "^0.4.1",
@@ -33,7 +34,7 @@
   "main": "index.js",
   "peerDependencies": {
     "gatsby": "^2.0.0-beta.29",
-    "netlify-cms": "^1.9.3"
+    "netlify-cms": "^2.0.6"
   },
   "repository": "https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-netlify-cms",
   "scripts": {

--- a/packages/gatsby-plugin-netlify-cms/src/cms.js
+++ b/packages/gatsby-plugin-netlify-cms/src/cms.js
@@ -1,7 +1,6 @@
 import CMS from "netlify-cms"
-import "netlify-cms/dist/cms.css"
 
-// eslint-disable-next-line no-undef
-if (NETLIFY_CMS_PREVIEW_STYLES_SET) {
-  CMS.registerPreviewStyle(`styles.css`)
-}
+/**
+ * The stylesheet output from the module at `stylesPath` will be at `cms.css`.
+ */
+CMS.registerPreviewStyle(`cms.css`)

--- a/packages/gatsby-plugin-netlify-cms/src/gatsby-browser.js
+++ b/packages/gatsby-plugin-netlify-cms/src/gatsby-browser.js
@@ -1,6 +1,6 @@
 import netlifyIdentityWidget from "netlify-identity-widget"
 
-exports.onInitialClientRender = (_, { enableIdentityWidget }) => {
+exports.onInitialClientRender = (_, { enableIdentityWidget = true }) => {
   if (enableIdentityWidget) {
     netlifyIdentityWidget.on(`init`, user => {
       if (!user) {

--- a/packages/gatsby-plugin-netlify-cms/src/gatsby-node.js
+++ b/packages/gatsby-plugin-netlify-cms/src/gatsby-node.js
@@ -1,7 +1,8 @@
 import path from "path"
-import { get, mapValues, isPlainObject, trim, pickBy } from "lodash"
+import { get, mapValues, isPlainObject, trim } from "lodash"
 import webpack from "webpack"
 import HtmlWebpackPlugin from "html-webpack-plugin"
+import HtmlWebpackExcludeAssetsPlugin from "html-webpack-exclude-assets-plugin"
 import MiniCssExtractPlugin from "mini-css-extract-plugin"
 import UglifyJsPlugin from "uglifyjs-webpack-plugin"
 import FriendlyErrorsPlugin from "friendly-errors-webpack-plugin"
@@ -34,13 +35,7 @@ function deepMap(obj, fn) {
 
 exports.onCreateWebpackConfig = (
   { store, stage, getConfig, plugins },
-  {
-    modulePath,
-    stylesPath,
-    publicPath = `admin`,
-    enableIdentityWidget = true,
-    htmlTitle = `Content Manager`,
-  }
+  { modulePath, publicPath = `admin`, enableIdentityWidget = true, htmlTitle = `Content Manager` },
 ) => {
   if ([`develop`, `build-javascript`].includes(stage)) {
     const gatsbyConfig = getConfig()
@@ -48,20 +43,13 @@ exports.onCreateWebpackConfig = (
     const publicPathClean = trim(publicPath, `/`)
     const config = {
       ...gatsbyConfig,
-      mode: `none`,
-      /**
-       * Two entries, one for the core CMS styles, and a `styles` entry, the
-       * output of which will be applied to the preview pane iframe only. Use
-       * `pickBy` to filter out empty entries.
-       */
-      entry: pickBy({
+      entry: {
         cms: [
           `${__dirname}/cms.js`,
           modulePath,
           enableIdentityWidget && `${__dirname}/cms-identity.js`,
         ].filter(p => p),
-        styles: stylesPath,
-      }),
+      },
       output: {
         path: path.join(program.directory, `public`, publicPathClean),
       },
@@ -122,20 +110,22 @@ exports.onCreateWebpackConfig = (
         new HtmlWebpackPlugin({
           title: htmlTitle,
           chunks: [`cms`],
+          excludeAssets: [/cms.css/],
         }),
 
         /**
-         * Set flag if custom styles path is added to plugin options.
+         * Exclude CSS from index.html, as any imported styles are assumed to be
+         * targeting the editor preview pane. Uses `excludeAssets` option from
+         * `HtmlWebpackPlugin` config.
          */
-        plugins.define({
-          NETLIFY_CMS_PREVIEW_STYLES_SET: !!stylesPath,
-        }),
+        new HtmlWebpackExcludeAssetsPlugin(),
       ].filter(p => p),
 
       /**
-       * Remove common chunks style optimizations from Gatsby's default config,
-       * they cause issues for our pre-bundled code.
+       * Remove mode and common chunks style optimizations from Gatsby's default
+       * config, they cause issues for our pre-bundled code.
        */
+      mode: `none`,
       optimization: {},
     }
     webpack(config).run()

